Ok, I've made some updates to this PR, but I'm still running into some errors. I generally know what's going wrong, but I'm not sure how to fix them, so a second pair of eyes from @helrond or @HaSistrunk could be useful.

FAILED tests/test_data_helpers.py::TestDataHelpers::test_get_expression - KeyError: 'expression'
One of the date objects this test is running against doesn't have an expression. Date objects without an expression don't include the expression at all, as far as I'm aware. The helper is failing because it can't find the key, but it has logic for what to do if the expression doesn't exist, which doesn't seem to be working.

FAILED tests/test_data_helpers.py::TestDataHelpers::test_text_in_note - AttributeError: 'list' object has no attribute 'strip'
The data helpers as written are trying to strip and lowercase note content. The example [here](FAILED tests/test_data_helpers.py::TestDataHelpers::test_text_in_note - AttributeError: 'list' object has no attribute 'strip') suggests that the content is a list. If I remove the list and just have a string from the single part note, it uncovers a different error.

FAILED tests/test_data_helpers.py::TestDataHelpers::test_get_note_text - AssertionError: <generator object get_note_text.<locals>.<genexpr> at 0x10cab70d0> != 'materials are restricted'
Trying to test an object against a string. Not sure how to get it to correctly test the right thing.

FAILED tests/test_data_helpers.py::TestDataHelpers::test_is_restricted - requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=4567): Max retries exceeded with url: /repositori...
Not sure what's going on with this one.

--- a/rac_aspace/data_helpers.py
+++ b/rac_aspace/data_helpers.py
@@ -43,9 +43,9 @@ def get_note_text(note):
         return content
 
     if note.jsonmodel_type == "note_singlepart":
-        content = note.content.strip("]['").split(', ')
+        content = note.content
     elif note.jsonmodel_type == "note_index":
-        content = note.items.strip("]['").split(', ')
+        content = note.items
     else:
         content = (parse_subnote(sn) for sn in note.subnotes)
     return content
@@ -65,7 +65,8 @@ def text_in_note(note, query_string):
     CONFIDENCE_RATIO = 97
     """int: Minimum confidence ratio to match against."""
     note_content = get_note_text(note)
-    ratio = fuzz.token_sort_ratio(note_content.lower(), query_string.lower())
+    ratio = fuzz.token_sort_ratio(
+        " ".join([n.lower() for n in note_content]), query_string.lower())
     return (True if ratio > CONFIDENCE_RATIO else False)
 
 

--- a/rac_aspace/data_helpers.py
+++ b/rac_aspace/data_helpers.py
@@ -180,12 +180,13 @@ def get_expression(date):
     Returns:
         str: a date expression for the date object.
     """
-    if date.expression:
+    try:
         return date.expression
-    if date.end:
-        return "{0}-{1}".format(date.begin, date.end)
-    else:
-        return date.begin
+    except KeyError:
+        if date.end:
+            return "{0}-{1}".format(date.begin, date.end)
+        else:
+            return date.begin
 
 
 def associated_objects(top_container):

--- a/rac_aspace/data_helpers.py
+++ b/rac_aspace/data_helpers.py
@@ -180,13 +180,12 @@ def get_expression(date):
     Returns:
         str: a date expression for the date object.
     """
-    try:
+    if date.expression:
         return date.expression
-    except KeyError:
-        if date.end:
-            return "{0}-{1}".format(date.begin, date.end)
-        else:
-            return date.begin
+    if date.end:
+        return "{0}-{1}".format(date.begin, date.end)
+    else:
+        return date.begin
 
 
 def associated_objects(top_container):

--- a/tests/test_data_helpers.py
+++ b/tests/test_data_helpers.py
@@ -39,14 +39,12 @@ class TestDataHelpers(unittest.TestCase):
         Returns:
             bool: True if the sort ratio is greater than or equal to 97.
         """
-        CONFIDENCE_RATIO = 97
         query_string = "New York Mets"
-        """int: Minimum confidence ratio to match against."""
         with open(os.path.join("fixtures", "note_single.json"), "r") as json_file:
             data = json.load(json_file)
             note = wrap_json_object(data)
             result = text_in_note(note, query_string)
-            self.assertTrue(result >= CONFIDENCE_RATIO)
+            self.assertTrue(result)
 
     # def test_get_locations(self):
         """

